legend-engine-xt-relationalStore-grammar/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperRelationalBuilder.java

```diff
@@ -961,6 +963,20 @@ private static PropertyMapping processRelationalPropertyMapping(RelationalProper
             Assert.assertTrue(eMap != null, () -> "Can't find enumeration mapping '" + propertyMapping.enumMappingId + "'");
             res = res._transformer(eMap);
         }
+        try
+        {
+            Pair<String, String> compareRelationalPropertyMappingTypes = (Pair<String, String>) Root_meta_relational_mapping_getIncompatibleTypesInRelationalPropertyMapping_RelationalPropertyMapping_$0_1$__Pair_$0_1$_(res, context.pureModel.getExecutionSupport());
+
+            if (compareRelationalPropertyMappingTypes != null)
+            {
+                context.pureModel.addWarnings(Lists.mutable.with(new Warning(SourceInformationHelper.fromM3SourceInformation(res.getSourceInformation()), "Types of class property: " + compareRelationalPropertyMappingTypes._first() + " and its mapped relational property: " + compareRelationalPropertyMappingTypes._second() + " are incompatible")));
```

Comment:
we need to include (Class name.property name) in the error message too, otherwise it will be hard to figure out the wrong property